ephios/user_management/templates/user_management/userprofile_form.html

```diff
@@ -21,10 +21,64 @@ <h1>{% trans "Add new user" %}</h1>
     </div>
     <form method="post" class="form">
         {% csrf_token %}
-        {% bootstrap_form form %}
+        {% bootstrap_form userprofile_form %}
+        {% load formset_tags %}
+
+        <div class="formset" data-formset data-formset-prefix="{{ qualification_formset.prefix }}">
+            {{ qualification_formset.management_form }}
+
+            <div data-formset-body>
+                <!-- New forms will be inserted in here -->
+                {% for form in qualification_formset %}
+                    <div class="card mb-3" data-formset-form>
+                        <div class="card-body clearfix">
+                            {{ form.id }}
+                            <button class="btn btn-link float-right pt-4" type="button" data-formset-delete-button><span
+                                class="fas fa-trash-alt"></span> {% trans "Delete" %}</button>
+                            <div class="form-row">
+                                <div class="col">
+                                    {% bootstrap_field form.qualification %}
```

Comment:
I think it makes no sense for an existing grant to be modified. Changing the qualification would better be a *new* grant. Changing the expires field might be required, but you could create a new grant still, then. Deletion is required on the other hand.